### DISCUSSION TITLE
supported temperature controllers?

### DISCUSSION THREAD

**Discussion by ashrond** [thorrak/fermentrack#774] [Q&A]
I am looking for a natively supported temperature controller recommendation. Failing that, can we just get native zigbee support so I could grab a zigbee outlet and give fermentrack control over it for the heater?

**Comment by thorrak** [comment]
Hey there! 

Fermentrack is designed to directly support BrewPi temperature controllers with a specific focus on [BrewPi-ESP](https://github.com/thorrak/brewpi-esp8266). The BrewPi implements the temperature control algorithm and actually handles the process of switching relays on/off to control cooling/heating. The latest enhancements to BrewPi-ESP allows an ESP-32-based controller to toggle [TPLink Kasa](https://www.amazon.com/TP-LINK-HS103P2-Required-Google-Assistant/dp/B07B8W2KHZ/) switches on/off rather than hard-wired relays, as well as leverage a Tilt as a wireless temperature sensor rather than a hard-wired DS18b20. 

Is that what you're looking for, or were you thinking of something different?

**Reply by ashrond** [reply]
Hello Thorrak,
Thank you for getting back to me so quickly, it is much appreciated!
sorry about pinging you here and on reddit, but thank you for the response at both.
May i ask, how does the ESP32 integrate with the Tilt sensor, if that is being monitored by fermentrack?

**Reply by thorrak** [reply]
No worries at all -- this is definitely the better place to get my attention!

The Tilt sensor broadcasts a bluetooth signal that can be read by multiple devices at once. This signal can be picked up simultaneously by any combination of a Raspberry Pi running Fermentrack, a [TiltBridge](https://www.tiltbridge.com/), or a [BrewPi-ESP](https://github.com/thorrak/brewpi-esp8266) running on an ESP32. The way the sensor is integrated changes depending on what device you use it with

**BrewPi-ESP**

The BrewPi-ESP firmware can - when running on an ESP32 - leverage a Tilt Pro's _temperature_ readings as a replacement for a hard-wired beer temperature sensor. This is great if you ferment in a vessel that doesn't have an integrated thermowell - but also removes one component that you would otherwise need to solder up to work with a BrewPi.

At the moment nothing is done with the received gravity information, but this is something that I'm looking to remedy with [Fermentrack 2](https://www.fermentrack.net/). 


**Fermentrack**

Fermentrack running on a Raspberry Pi can use the Pi's bluetooth radio to read a Tilt's broadcasts and capture the Tilt's gravity/temperature information. This data is not used in temperature _control_, but can be logged alongside your temperature control data to see how your beer's fermentation is progressing.


**TiltBridge**

A TiltBridge is a project of mine that leverages an ESP32 to read a Tilt's Bluetooth signal  and "bridge" the data to a number of internet endpoints using WiFi. One of the endpoints it can send to is Fermentrack. This allows Fermentrack to use the Tilt's gravity data as if it was connected natively to the Pi.

The use of a TiltBridge makes it so that you don't have to keep your Pi next to your fermenter and can still capture/log gravity information.
